### PR TITLE
xxx.h, xxx.{cc,cu}, xxx_test.{cc,cu} in the same directory

### LINKED ISSUE #2209: xxx.h, xxx.{cc,cu}, xxx_test.{cc,cu} in the same directory
As we are adopting the [new building system](https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/generic.cmake), whose design doc is at [here](https://github.com/PaddlePaddle/Paddle/pull/2187), we should build each unit test into a standalone executable binary file.  So we don't have to put all unit tests of a C++ package into a `test` directory. Instead; we can put all three files of each module together in the same directory.

### PR DESCRIPTION
**Pull request by Canpio**
fix #2209

### CODE REVIEW

File: paddle/majel/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,13 @@
-cc_library(place SRCS place.cc)
-cc_library(ddim SRCS ddim.cc)
+cc_library(place SRCS place/place.cc)
+cc_library(ddim SRCS ddim/ddim.cc)
 
 if(WITH_TESTING)
```
Comment (Xreki) [inline paddle/majel/CMakeLists.txt]: @gangliao  `WITH_TESTING` 和 `WITH_GPU`等判断是不是可以放在`cc_test`和`nv_test`的定义里面，这样`CMakeLists.txt`看起来会更简洁一些。

Comment (wangkuiyi) [inline paddle/majel/CMakeLists.txt]: I created a new issue https://github.com/PaddlePaddle/Paddle/issues/2259 to record @Xreki 's suggestion.

File: paddle/majel/ddim/ddim.cc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "paddle/majel/ddim.h"
+#include "paddle/majel/ddim/ddim.h"
```
Comment (wangkuiyi) [inline paddle/majel/ddim/ddim.cc]: I tend to think that we don't need to move each C++ module into a directory.  In C++, a module is defined by a .cc file, optionally with interface declaration in a .h file.  We can describe a C++ module by a CMake rule, e.g., `cc_library(ddim SRCS ddim.cc)`. So it seems overkilling to move each C++ module into a directory.

Comment (Canpio) [inline paddle/majel/ddim/ddim.cc]: I agree that current CMake rule in Paddle majel with `cc_library()` and `cc_test()` has made module describing quite clearly, therefore simple module like `ddim` and `place` is not necessery to moved into a directory. However, if a module is really complicated, we should still gather related files up in a subdirectory.

This is my personal understanding, I'm not sure whether it is right ...

Comment (wangkuiyi) [inline paddle/majel/ddim/ddim.cc]: I think you are right for the complicated case, e.g., paddle/parameter.  For this case, let's move ddim back to `paddle/majel`.

Comment (Canpio) [inline paddle/majel/ddim/ddim.cc]: Got it. I'm going to close this PR.

**Pull request closed without merging**